src/ai/defaultai.cc

```diff
@@ -6777,6 +6846,11 @@ void DefaultAI::gain_building(Widelands::Building& b, const bool found_on_load)
 		}
 		if (target_bo.type == BuildingObserver::Type::kWarehouse) {
 			++numof_warehouses_in_const_;
+			if (!found_on_load) {
```

Comment:
setting the task scheduler here does not make sense as we just gained a condtructionsite which isn't considered in the checking for distances as only finished warehouses are counted. 
so either we could afford to skip this, as the distances are calculated every 15 seconds gametime
or we need to move it down to line 6941 ff

Reply:
good catch!